lib/trento/application/integration/discovery/policies/sap_system_policy.ex

```diff
@@ -82,22 +87,26 @@ defmodule Trento.Integration.Discovery.SapSystemPolicy do
          },
          host_id
        ) do
-    Enum.map(instances, fn {_, instance} ->
-      instance_number = parse_instance_number(instance)
-
-      RegisterApplicationInstance.new(%{
-        sid: sid,
-        tenant: tenant,
-        db_host: db_host,
-        instance_number: instance_number,
-        instance_hostname: parse_instance_hostname(instance, instance_number),
-        features: parse_features(instance, instance_number),
-        http_port: parse_http_port(instance, instance_number),
-        https_port: parse_https_port(instance, instance_number),
-        start_priority: parse_start_priority(instance, instance_number),
-        host_id: host_id,
-        health: parse_instance_health(instance, instance_number)
-      })
+    Enum.map(instances, fn instance ->
+      case parse_instance_number(instance) do
```

Comment:
if we return value or nil from this function, the case is not needed since ecto would guard against blank/nil values (the field is required in the defcommand)